File: API_SC_01_Marco2023/G2/Algoritmo-Luana/meu_dinheirinho.js
import {question} from 'readline-sync'

function main(){
    const valor_base = Number(question('Valor base hora/aula(R$): '))
    const professor = question('Nome do professor: ')
    const h_semanais = Number(question('Horas semanais: '))
    const qualificacao = question('Qualificacao(E/M/D): ')
    const tempo_experiencia = Number(question('Tempo de experiencia(meses): '))
    const qtd_filhos = Number(question('Quant. de filhos(menores de 6 anos): '))
    const plano_saude = Number(question('Valor do plano de saude(R$): '))
    
    const valor_por_especializacao = calcular_valor_por_especializacao(valor_base, qualificacao)
    const valor_experiencia = calcular_valor_experiencia(valor_base, tempo_experiencia)
    const valor_total_hora_aula = valor_por_especializacao + valor_experiencia
    const salario_semanal = calcular_salario_semanal(h_semanais, valor_total_hora_aula)
    const salario_mensal = calcular_salario_mensal(salario_semanal)
    const auxilio_creche = calcular_auxilio_creche(qtd_filhos, salario_mensal)
    
    console.log('-------Contracheque--------')
    console.log('Professor(a): ', professor)
    console.log(`Valor Hora/Professor: R$ ${valor_total_hora_aula}`)
    console.log(`Salário base semanal: R$ ${salario_semanal}`)
    console.log('- GANHOS ---------')
    console.log(`Salário base mensal: R$ ${salario_mensal.toFixed(2)}`)
    console.log(`Auxílio creche: R$ ${auxilio_creche.toFixed(2)}`)

}

function calcular_valor_por_especializacao(base, qualificacao){
    if(qualificacao === 'E'){
        return (base * 0.20) + base
    }else if(qualificacao === 'M'){
        return (base * 0.30) + base
    }else if(qualificacao === 'D'){
        return (base * 0.50) + base
    }
}

function calcular_valor_experiencia(base, tempo){
    if(tempo === 6){
        return base * 0.05
    }else if(tempo > 6){
        const anos = tempo/12
        return (base * 0.05 * anos)
    }else{
        return 0
    }
}

function calcular_salario_semanal(horas_semanal, valor_uma_hora){
    return horas_semanal * valor_uma_hora
}

function calcular_salario_mensal(salario_semanal){
    return salario_semanal * 4.5
}

function calcular_auxilio_creche(qtd_filhos, salario){
    return 700 * qtd_filhos
}

main()

File: API_SC_01_Marco2023/G2/Luiz David/meu_dinheirinho.py
from auxiliar_functions import *


def main():
    # solicitar dados
    valor_hr = get_valid_value("valor da hora/aula na universidade: ")
    nome_professor = input("nome: ")
    qtd_hr_semanais = get_valid_value("quantidade de horas semanais: ")
    qualificacao = get_valid_qualification("sua formação profissional: ")
    qtd_filhos = get_valid_value("quantidade de filhos menores de 6 anos: ")
    valor_plano_saude = get_valid_value("valor do seu plano de saúde R$:")
    anos_experiencia = get_valid_value("quantidade de anos de experiencia: ")

    salario_base_mensal = calcular_salario_mensal(
        valor_hr, qtd_hr_semanais, qualificacao, anos_experiencia)
    
    auxilio_creche, valor_ressarcimento, valor_auxilio_combustivel = calcular_auxilios(qtd_filhos, valor_plano_saude, qtd_hr_semanais)
    
    inss = calcular_inss(salario_base_mensal)
    ir = calcular_ir(salario_base_mensal)
    
    salario_bruto = (auxilio_creche + valor_auxilio_combustivel + valor_ressarcimento)
    desconto_total = inss + ir
    salario_liquido = salario_bruto - desconto_total
    
    print(f"\nprofessor: {nome_professor}")
    print(f"valor da hora: {valor_hr:.2f}")
    print(f"salário base semanal sem auxilio: R$ {(salario_base_mensal / 4.5):.2f}")
    print(f"\nsalário base mensal sem auxilio: R$ {salario_base_mensal:.2f}")
    print(f"auxilio creche: R$ {auxilio_creche:.2f}")
    print(f"reassarcimento do plano de saúde: R$ {valor_ressarcimento:.2f}")
    print(f"auxílio combustível: R$ {valor_auxilio_combustivel:.2f}")
    print(f"salário bruto: R$ {salario_bruto:.2f}")
    print(f"desconto ir: R$ {ir:.2f}")
    print(f"desconto inss: R$ {inss:.2f}")
    print(f"desconto total: R$ {desconto_total}")
    print(f"salário líquido: R$ {salario_liquido:.2f}")


def calcular_salario_mensal(valor_hr, qtd_hrs, qualificacao, anos_experiencia):
    # calcular salario sem beneficios e descontos
    salario = valor_hr * qtd_hrs

    if qualificacao == "especilização":
        bonus_qualificacao = salario * 0.2

    elif qualificacao == "mestrado":
        bonus_qualificacao = salario * 0.3

    elif qualificacao == "doutorado":
        bonus_qualificacao = salario * 0.5

    # calcular anos de experiencia
    if anos_experiencia > 1:
        valor_experiencia = valor_hr * 0.5
        salario_base_mensal += valor_experiencia


    salario_base_mensal = (salario + bonus_qualificacao) * 4.5

    return salario_base_mensal


def calcular_inss(salario_base_mensal):
    # calcular previdencia social

    if salario_base_mensal <= 1320:
        inss = salario_base_mensal * (7.5 / 100)

    elif salario_base_mensal > 1320 and salario_base_mensal <= 2500:
        inss = salario_base_mensal * (9 / 100)

    elif salario_base_mensal > 2500 and salario_base_mensal <= 3900:
        inss = salario_base_mensal * (12 / 100)

    elif salario_base_mensal > 3900 and salario_base_mensal <= 7500:
        inss = salario_base_mensal * (14 / 100)

    elif salario_base_mensal > 7500:
        inss = salario_base_mensal * (16 / 100)

    return inss


def calcular_ir(salario_base_mensal):
    # calcular ir

    if salario_base_mensal <= 5000:
        ir = 0

    else:
        ir = salario_base_mensal * (27.5 / 100)

    return ir


def calcular_auxilios(qtd_filhos, valor_plano_saude, qtd_hr_semanais):
    qtd_aulas = qtd_hr_semanais * 4.5

    # calcular salario com bonus sem descontos
    auxilio_creche = qtd_filhos * 700

    # calcular quanto ressarcer do plano de saúde
    valor_ressarcimento = valor_plano_saude / 2

    if valor_plano_saude > 1000:
        valor_ressarcimento = 500
    
    # calcular valor do auxilio combustível
    import math

    valor_auxilio_combustivel = math.floor(qtd_aulas / 8) * 30

    return auxilio_creche, valor_ressarcimento, valor_auxilio_combustivel


main()




